Async client example does not compile

If I create a project with the rumqttc example code and try to compile it (under tag R3 or head of master), rumqttc fails to compile with:

```plain
   Compiling rumqttc v0.3.0 (https://github.com/detly/rumqtt.git?tag=R3#b8688681)
error[E0308]: mismatched types
   --> /home/jason/.cargo/git/checkouts/rumqtt-38df926bebc19038/b868868/rumqttc/src/eventloop.rs:292:40
    |
292 |             Network::new(WsStream::new(socket), options.max_incoming_packet_size)
    |                                        ^^^^^^ expected struct `async_tungstenite::WebSocketStream`, found struct `WebSocketStream`
    |
    = note: expected struct `async_tungstenite::WebSocketStream<_>`
               found struct `WebSocketStream<async_tungstenite::stream::Stream<TokioAdapter<tokio::net::TcpStream>, TokioAdapter<tokio_rustls::client::TlsStream<tokio::net::TcpStream>>>>`
    = note: perhaps two different versions of crate `async_tungstenite` are being used?

error[E0277]: the trait bound `WsStream<_>: AsyncRead` is not satisfied
   --> /home/jason/.cargo/git/checkouts/rumqtt-38df926bebc19038/b868868/rumqttc/src/eventloop.rs:292:26
    |
292 |             Network::new(WsStream::new(socket), options.max_incoming_packet_size)
    |                          ^^^^^^^^^^^^^^^^^^^^^ the trait `AsyncRead` is not implemented for `WsStream<_>`
    | 
   ::: /home/jason/.cargo/git/checkouts/rumqtt-38df926bebc19038/b868868/rumqttc/src/framed.rs:23:29
    |
23  |     pub fn new(socket: impl N + 'static, max_incoming_size: usize) -> Network {
    |                             - required by this bound in `Network::new`
    |
    = note: required because of the requirements on the impl of `N` for `WsStream<_>`

error[E0277]: the trait bound `WsStream<_>: AsyncWrite` is not satisfied
   --> /home/jason/.cargo/git/checkouts/rumqtt-38df926bebc19038/b868868/rumqttc/src/eventloop.rs:292:26
    |
292 |             Network::new(WsStream::new(socket), options.max_incoming_packet_size)
    |                          ^^^^^^^^^^^^^^^^^^^^^ the trait `AsyncWrite` is not implemented for `WsStream<_>`
    | 
   ::: /home/jason/.cargo/git/checkouts/rumqtt-38df926bebc19038/b868868/rumqttc/src/framed.rs:23:29
    |
23  |     pub fn new(socket: impl N + 'static, max_incoming_size: usize) -> Network {
    |                             - required by this bound in `Network::new`
    |
    = note: required because of the requirements on the impl of `N` for `WsStream<_>`

error[E0308]: mismatched types
   --> /home/jason/.cargo/git/checkouts/rumqtt-38df926bebc19038/b868868/rumqttc/src/eventloop.rs:308:40
    |
308 |             Network::new(WsStream::new(socket), options.max_incoming_packet_size)
    |                                        ^^^^^^ expected struct `async_tungstenite::WebSocketStream`, found struct `WebSocketStream`
    |
    = note: expected struct `async_tungstenite::WebSocketStream<_>`
               found struct `WebSocketStream<async_tungstenite::stream::Stream<TokioAdapter<tokio::net::TcpStream>, TokioAdapter<tokio_rustls::client::TlsStream<tokio::net::TcpStream>>>>`
    = note: perhaps two different versions of crate `async_tungstenite` are being used?

error[E0277]: the trait bound `WsStream<_>: AsyncRead` is not satisfied
   --> /home/jason/.cargo/git/checkouts/rumqtt-38df926bebc19038/b868868/rumqttc/src/eventloop.rs:308:26
    |
308 |             Network::new(WsStream::new(socket), options.max_incoming_packet_size)
    |                          ^^^^^^^^^^^^^^^^^^^^^ the trait `AsyncRead` is not implemented for `WsStream<_>`
    | 
   ::: /home/jason/.cargo/git/checkouts/rumqtt-38df926bebc19038/b868868/rumqttc/src/framed.rs:23:29
    |
23  |     pub fn new(socket: impl N + 'static, max_incoming_size: usize) -> Network {
    |                             - required by this bound in `Network::new`
    |
    = note: required because of the requirements on the impl of `N` for `WsStream<_>`

error[E0277]: the trait bound `WsStream<_>: AsyncWrite` is not satisfied
   --> /home/jason/.cargo/git/checkouts/rumqtt-38df926bebc19038/b868868/rumqttc/src/eventloop.rs:308:26
    |
308 |             Network::new(WsStream::new(socket), options.max_incoming_packet_size)
    |                          ^^^^^^^^^^^^^^^^^^^^^ the trait `AsyncWrite` is not implemented for `WsStream<_>`
    | 
   ::: /home/jason/.cargo/git/checkouts/rumqtt-38df926bebc19038/b868868/rumqttc/src/framed.rs:23:29
    |
23  |     pub fn new(socket: impl N + 'static, max_incoming_size: usize) -> Network {
    |                             - required by this bound in `Network::new`
    |
    = note: required because of the requirements on the impl of `N` for `WsStream<_>`

error: aborting due to 6 previous errors

Some errors have detailed explanations: E0277, E0308.
For more information about an error, try `rustc --explain E0277`.
error: could not compile `rumqttc`

To learn more, run the command again with --verbose.
warning: build failed, waiting for other jobs to finish...
error: build failed
```

cargo 1.49.0, rustc 1.49.0 under Ubuntu 20.10.

Just trying to compile rumqttc will exhibit this by the way. No example code required. Bumping these dependencies in rumqttc gets rid of the remaining "perhaps two different versions" etc. problem:

```toml
tokio-rustls = "0.22"
async-tungstenite = { version = "0.11.0", default-features = false, features = ["tokio-rustls"] }
```

But now I have:

```plain
   Compiling rumqttc v0.3.0 (/home/jason/Code/rumqtt/rumqttc)
error[E0277]: the trait bound `tokio_rustls::client::TlsStream<tokio::net::TcpStream>: AsyncRead` is not satisfied
   --> rumqttc/src/eventloop.rs:278:26
    |
278 |             Network::new(socket, options.max_incoming_packet_size)
    |                          ^^^^^^ the trait `AsyncRead` is not implemented for `tokio_rustls::client::TlsStream<tokio::net::TcpStream>`
    | 
   ::: rumqttc/src/framed.rs:23:29
    |
23  |     pub fn new(socket: impl N + 'static, max_incoming_size: usize) -> Network {
    |                             - required by this bound in `Network::new`
    |
    = note: required because of the requirements on the impl of `N` for `tokio_rustls::client::TlsStream<tokio::net::TcpStream>`

error[E0277]: the trait bound `tokio_rustls::client::TlsStream<tokio::net::TcpStream>: AsyncWrite` is not satisfied
   --> rumqttc/src/eventloop.rs:278:26
    |
278 |             Network::new(socket, options.max_incoming_packet_size)
    |                          ^^^^^^ the trait `AsyncWrite` is not implemented for `tokio_rustls::client::TlsStream<tokio::net::TcpStream>`
    | 
   ::: rumqttc/src/framed.rs:23:29
    |
23  |     pub fn new(socket: impl N + 'static, max_incoming_size: usize) -> Network {
    |                             - required by this bound in `Network::new`
    |
    = note: required because of the requirements on the impl of `N` for `tokio_rustls::client::TlsStream<tokio::net::TcpStream>`

error[E0277]: the trait bound `WsStream<async_tungstenite::stream::Stream<TokioAdapter<tokio::net::tcp::stream::TcpStream>, TokioAdapter<tokio_rustls::client::TlsStream<tokio::net::tcp::stream::TcpStream>>>>: AsyncRead` is not satisfied
   --> rumqttc/src/eventloop.rs:292:26
    |
292 |             Network::new(WsStream::new(socket), options.max_incoming_packet_size)
    |                          ^^^^^^^^^^^^^^^^^^^^^ the trait `AsyncRead` is not implemented for `WsStream<async_tungstenite::stream::Stream<TokioAdapter<tokio::net::tcp::stream::TcpStream>, TokioAdapter<tokio_rustls::client::TlsStream<tokio::net::tcp::stream::TcpStream>>>>`
    | 
   ::: rumqttc/src/framed.rs:23:29
    |
23  |     pub fn new(socket: impl N + 'static, max_incoming_size: usize) -> Network {
    |                             - required by this bound in `Network::new`
    |
    = note: required because of the requirements on the impl of `N` for `WsStream<async_tungstenite::stream::Stream<TokioAdapter<tokio::net::tcp::stream::TcpStream>, TokioAdapter<tokio_rustls::client::TlsStream<tokio::net::tcp::stream::TcpStream>>>>`

error[E0277]: the trait bound `WsStream<async_tungstenite::stream::Stream<TokioAdapter<tokio::net::tcp::stream::TcpStream>, TokioAdapter<tokio_rustls::client::TlsStream<tokio::net::tcp::stream::TcpStream>>>>: AsyncWrite` is not satisfied
   --> rumqttc/src/eventloop.rs:292:26
    |
292 |             Network::new(WsStream::new(socket), options.max_incoming_packet_size)
    |                          ^^^^^^^^^^^^^^^^^^^^^ the trait `AsyncWrite` is not implemented for `WsStream<async_tungstenite::stream::Stream<TokioAdapter<tokio::net::tcp::stream::TcpStream>, TokioAdapter<tokio_rustls::client::TlsStream<tokio::net::tcp::stream::TcpStream>>>>`
    | 
   ::: rumqttc/src/framed.rs:23:29
    |
23  |     pub fn new(socket: impl N + 'static, max_incoming_size: usize) -> Network {
    |                             - required by this bound in `Network::new`
    |
    = note: required because of the requirements on the impl of `N` for `WsStream<async_tungstenite::stream::Stream<TokioAdapter<tokio::net::tcp::stream::TcpStream>, TokioAdapter<tokio_rustls::client::TlsStream<tokio::net::tcp::stream::TcpStream>>>>`

error[E0277]: the trait bound `WsStream<async_tungstenite::stream::Stream<TokioAdapter<tokio::net::tcp::stream::TcpStream>, TokioAdapter<tokio_rustls::client::TlsStream<tokio::net::tcp::stream::TcpStream>>>>: AsyncRead` is not satisfied
   --> rumqttc/src/eventloop.rs:308:26
    |
308 |             Network::new(WsStream::new(socket), options.max_incoming_packet_size)
    |                          ^^^^^^^^^^^^^^^^^^^^^ the trait `AsyncRead` is not implemented for `WsStream<async_tungstenite::stream::Stream<TokioAdapter<tokio::net::tcp::stream::TcpStream>, TokioAdapter<tokio_rustls::client::TlsStream<tokio::net::tcp::stream::TcpStream>>>>`
    | 
   ::: rumqttc/src/framed.rs:23:29
    |
23  |     pub fn new(socket: impl N + 'static, max_incoming_size: usize) -> Network {
    |                             - required by this bound in `Network::new`
    |
    = note: required because of the requirements on the impl of `N` for `WsStream<async_tungstenite::stream::Stream<TokioAdapter<tokio::net::tcp::stream::TcpStream>, TokioAdapter<tokio_rustls::client::TlsStream<tokio::net::tcp::stream::TcpStream>>>>`

error[E0277]: the trait bound `WsStream<async_tungstenite::stream::Stream<TokioAdapter<tokio::net::tcp::stream::TcpStream>, TokioAdapter<tokio_rustls::client::TlsStream<tokio::net::tcp::stream::TcpStream>>>>: AsyncWrite` is not satisfied
   --> rumqttc/src/eventloop.rs:308:26
    |
308 |             Network::new(WsStream::new(socket), options.max_incoming_packet_size)
    |                          ^^^^^^^^^^^^^^^^^^^^^ the trait `AsyncWrite` is not implemented for `WsStream<async_tungstenite::stream::Stream<TokioAdapter<tokio::net::tcp::stream::TcpStream>, TokioAdapter<tokio_rustls::client::TlsStream<tokio::net::tcp::stream::TcpStream>>>>`
    | 
   ::: rumqttc/src/framed.rs:23:29
    |
23  |     pub fn new(socket: impl N + 'static, max_incoming_size: usize) -> Network {
    |                             - required by this bound in `Network::new`
    |
    = note: required because of the requirements on the impl of `N` for `WsStream<async_tungstenite::stream::Stream<TokioAdapter<tokio::net::tcp::stream::TcpStream>, TokioAdapter<tokio_rustls::client::TlsStream<tokio::net::tcp::stream::TcpStream>>>>`

error[E0277]: the trait bound `tokio::net::TcpStream: tokio::io::async_read::AsyncRead` is not satisfied
  --> rumqttc/src/tls.rs:98:41
   |
98 |     let tls = connector.connect(domain, tcp).await?;
   |                                         ^^^ the trait `tokio::io::async_read::AsyncRead` is not implemented for `tokio::net::TcpStream`

error[E0277]: the trait bound `tokio::net::TcpStream: tokio::io::async_write::AsyncWrite` is not satisfied
  --> rumqttc/src/tls.rs:98:41
   |
98 |     let tls = connector.connect(domain, tcp).await?;
   |                                         ^^^ the trait `tokio::io::async_write::AsyncWrite` is not implemented for `tokio::net::TcpStream`

error[E0277]: the trait bound `tokio::net::TcpStream: tokio::io::async_read::AsyncRead` is not satisfied
  --> rumqttc/src/tls.rs:98:15
   |
98 |     let tls = connector.connect(domain, tcp).await?;
   |               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `tokio::io::async_read::AsyncRead` is not implemented for `tokio::net::TcpStream`
   |
   = note: required because of the requirements on the impl of `Future` for `tokio_rustls::Connect<tokio::net::TcpStream>`
   = note: required by `poll`

error[E0277]: the trait bound `tokio::net::TcpStream: tokio::io::async_write::AsyncWrite` is not satisfied
  --> rumqttc/src/tls.rs:98:15
   |
98 |     let tls = connector.connect(domain, tcp).await?;
   |               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `tokio::io::async_write::AsyncWrite` is not implemented for `tokio::net::TcpStream`
   |
   = note: required because of the requirements on the impl of `Future` for `tokio_rustls::Connect<tokio::net::TcpStream>`
   = note: required by `poll`

error: aborting due to 10 previous errors

For more information about this error, try `rustc --explain E0277`.
error: could not compile `rumqttc`

To learn more, run the command again with --verbose.
```

--- a/benchmarks/Cargo.toml
+++ b/benchmarks/Cargo.toml
@@ -10,12 +10,12 @@ edition = "2018"
 argh = "0.1"
 rumqttc = { path = "../rumqttc" }
 rumqttlog = { path = "../rumqttlog" }
-tokio = { version = "=0.2.22", features = ["full"] }
+tokio = { version = "1.0", features = ["full"] }
 futures = { version = "0.3", features = ["compat"] }	
-bytes = "0.5"
+bytes = "1.0"
 # paho-mqtt = { git = "https://github.com/eclipse/paho.mqtt.rust" }
 # paho-mqtt = "0.7"
-async-channel = "1.4"
+async-channel = "1.5"
 pretty_env_logger = "0.4"
 prost = "0.6"
 pprof = { version = "0.3", features = ["flamegraph", "protobuf"] }

--- a/benchmarks/mesh.rs
+++ b/benchmarks/mesh.rs
@@ -7,7 +7,7 @@ use rumqttlog::router::{Data};
 
 mod common;
 
-#[tokio::main(core_threads = 1)]
+#[tokio::main(worker_threads = 1)]
 async fn main() {
     pretty_env_logger::init();
 
@@ -28,7 +28,7 @@ async fn main() {
     read(tx).await;
 }
 
-#[tokio::main(core_threads = 1)]
+#[tokio::main(worker_threads = 1)]
 async fn start_router(mut router: Router) {
     router.start().await;
 }

--- a/benchmarks/pahoasync.rs
+++ b/benchmarks/pahoasync.rs
@@ -5,7 +5,7 @@ use std::error::Error;
 use mqtt::AsyncClient;
 use futures::executor::block_on;
 
-#[tokio::main(core_threads=2)]
+#[tokio::main(worker_threads=2)]
 async fn main() {
     let _o = start("paho-rust-async", 100, 100000);
 }

--- a/benchmarks/routernxn.rs
+++ b/benchmarks/routernxn.rs
@@ -37,7 +37,7 @@
 //     publisher_count: usize,
 // }
 //
-// #[tokio::main(core_threads = 8)]
+// #[tokio::main(worker_threads = 8)]
 // async fn main() {
 //     pretty_env_logger::init();
 //     let commandline: CommandLine = argh::from_env();
@@ -75,10 +75,10 @@
 //         }
 //     });
 //
-//     time::delay_for(Duration::from_secs(500)).await;
+//     time::sleep(Duration::from_secs(500)).await;
 // }
 //
-// #[tokio::main(core_threads = 1)]
+// #[tokio::main(worker_threads = 1)]
 // async fn start_router(mut router: Router) {
 //     router.start().await;
 // }

--- a/benchmarks/rumqttasync.rs
+++ b/benchmarks/rumqttasync.rs
@@ -11,7 +11,7 @@ mod common;
 #[global_allocator]
 static ALLOC: jemallocator::Jemalloc = jemallocator::Jemalloc;
 
-#[tokio::main(core_threads = 2)]
+#[tokio::main(worker_threads = 2)]
 async fn main() {
     // pretty_env_logger::init();
     let guard = pprof::ProfilerGuard::new(100).unwrap();
@@ -36,7 +36,7 @@ pub async fn start(id: &str, payload_size: usize, count: usize) -> Result<(), Bo
                 .unwrap();
         }
 
-        time::delay_for(Duration::from_secs(10)).await;
+        time::sleep(Duration::from_secs(10)).await;
     });
 
     let mut acks_count = 0;

--- a/benchmarks/rumqttasyncqos0.rs
+++ b/benchmarks/rumqttasyncqos0.rs
@@ -11,7 +11,7 @@ mod common;
 #[global_allocator]
 static ALLOC: jemallocator::Jemalloc = jemallocator::Jemalloc;
 
-#[tokio::main(core_threads = 2)]
+#[tokio::main(worker_threads = 2)]
 async fn main() {
     // pretty_env_logger::init();
     let guard = pprof::ProfilerGuard::new(100).unwrap();
@@ -42,7 +42,7 @@ pub async fn start(id: &str, payload_size: usize, count: usize) -> Result<(), Bo
             .publish("hello/benchmarks/world", qos, false, payload)
             .await
             .unwrap();
-        time::delay_for(Duration::from_secs(10)).await;
+        time::sleep(Duration::from_secs(10)).await;
     });
 
     let start = Instant::now();

--- a/mqtt4bytes/Cargo.toml
+++ b/mqtt4bytes/Cargo.toml
@@ -13,7 +13,7 @@ categories = ["network-programming", "no-std"]
 bench = false
 
 [dependencies]
-bytes = { version = "0.6", default-features = false }
+bytes = { version = "1.0", default-features = false }
 
 [dev-dependencies]
 pretty_assertions = "0.6.1"

--- a/mqtt5bytes/Cargo.toml
+++ b/mqtt5bytes/Cargo.toml
@@ -11,11 +11,11 @@ categories = ["network-programming"]
 # See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
 
 [dependencies]
-bytes = { version = "0.5", default-features = false }
+bytes = { version = "1.0", default-features = false }
 
 [dev-dependencies]
-tokio = { version = "0.2", features = ["full"] }
+tokio = { version = "1.0", features = ["full"] }
 pretty_assertions = "0.6.1"
-rand = "0.7"
+rand = "0.8"
 criterion = "0.3"
 

--- a/mqtt5bytes/examples/pubsub.rs
+++ b/mqtt5bytes/examples/pubsub.rs
@@ -4,7 +4,7 @@ use std::io;
 use tokio::io::{AsyncReadExt, AsyncWriteExt};
 use tokio::net::TcpStream;
 
-#[tokio::main(core_threads = 2)]
+#[tokio::main(worker_threads = 2)]
 async fn main() {
     let mut client = Client::new().await;
 

--- a/mqtt5bytes/examples/pubsub.rs
+++ b/mqtt5bytes/examples/pubsub.rs
@@ -3,8 +3,6 @@ use mqtt5bytes::{mqtt_read, Connect, Error, Packet, Publish, QoS, Subscribe};
 use std::io;
 use tokio::io::{AsyncReadExt, AsyncWriteExt};
 use tokio::net::TcpStream;
-use tokio::time;
-use tokio::time::Duration;
 
 #[tokio::main(core_threads = 2)]
 async fn main() {

--- a/mqtt5bytes/src/packets/mod.rs
+++ b/mqtt5bytes/src/packets/mod.rs
@@ -58,20 +58,6 @@ fn read_mqtt_string(stream: &mut Bytes) -> Result<String, Error> {
     }
 }
 
-/// After collecting enough bytes to frame a packet (to pass to packet's frame())
-/// , It's possible that content itself in the stream is wrong. Like expected
-/// packet id or qos not being present. In cases where `read_mqtt_string` or
-/// `read_mqtt_bytes` exhausted remaining length but packet framing expects to
-/// parse qos next, these pre checks will prevent `bytes` crashes
-fn required_minimum(stream: &mut Bytes, required: usize) -> Result<(), Error> {
-    let stream_len = stream.len();
-    if stream_len < required {
-        return Err(Error::MalformedPacket);
-    }
-
-    Ok(())
-}
-
 /// Serializes bytes to stream (including length)
 fn write_mqtt_bytes(stream: &mut BytesMut, bytes: &[u8]) {
     stream.put_u16(bytes.len() as u16);

--- a/mqtt5bytes/src/packets/subscribe.rs
+++ b/mqtt5bytes/src/packets/subscribe.rs
@@ -389,7 +389,7 @@ mod test {
     }
 
     fn sample2() -> Subscribe {
-        let mut filter = SubscribeFilter::new("hello/world".to_owned(), QoS::AtLeastOnce);
+        let filter = SubscribeFilter::new("hello/world".to_owned(), QoS::AtLeastOnce);
         Subscribe {
             pkid: 42,
             filters: vec![filter],

--- a/rumqttc/Cargo.toml
+++ b/rumqttc/Cargo.toml
@@ -12,12 +12,13 @@ categories = ["network-programming"]
 # See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
 
 [dependencies]
-tokio = { version = "0.3.2", features = ["full"] }
-bytes = "0.6"
+tokio = { version = "1.0", features = ["full"] }
+tokio-stream = "0.1"
+bytes = "1.0"
 webpki = "0.21"
-tokio-rustls = "0.20"
-async-tungstenite = { version = "0.10.0", default-features = false, features = ["tokio-rustls"] }
-ws_stream_tungstenite = { version = "0.4.0-beta.2", default-features = false, features = ["tokio_io"] }
+tokio-rustls = "0.22"
+async-tungstenite = { version = "0.11.0", default-features = false, features = ["tokio-rustls"] }
+ws_stream_tungstenite = { version = "0.4.0", default-features = false, features = ["tokio_io"] }
 mqtt4bytes = { path = "../mqtt4bytes", version = "0.4" }
 pollster = "0.2"
 async-channel = "1.5"
@@ -32,5 +33,5 @@ crossbeam-channel = "0.5"
 serde = {version = "1", features = ["derive"]}
 envy = "0.4"
 jsonwebtoken = "7"
-tokio = { version = "0.3.2", features = ["full", "macros"] }
+tokio = { version = "1.0", features = ["full", "macros"] }
 matches = "0.1.8"

--- a/rumqttc/README.md
+++ b/rumqttc/README.md
@@ -49,14 +49,14 @@ client.subscribe("hello/rumqtt", QoS::AtMostOnce).await.unwrap();
 task::spawn(async move {
     for i in 0..10 {
         client.publish("hello/rumqtt", QoS::AtLeastOnce, false, vec![i; i as usize]).await.unwrap();
-        time::delay_for(Duration::from_millis(100)).await;
+        time::sleep(Duration::from_millis(100)).await;
     }
 });
 
 loop {
     let notification = eventloop.poll().await.unwrap();
     println!("Received = {:?}", notification);
-    tokio::time::delay_for(Duration::from_secs(1)).await;
+    tokio::time::sleep(Duration::from_secs(1)).await;
 }
 ```
 

--- a/rumqttc/src/eventloop.rs
+++ b/rumqttc/src/eventloop.rs
@@ -7,11 +7,12 @@ use async_tungstenite::tokio::{connect_async, connect_async_with_tls_connector};
 use mqtt4bytes::*;
 use tokio::net::TcpStream;
 use tokio::select;
-use tokio::stream::{Stream, StreamExt};
 use tokio::time::{self, error::Elapsed, Instant, Sleep};
+use tokio_stream::{Stream, StreamExt};
 use ws_stream_tungstenite::WsStream;
 
 use std::io;
+use std::pin::Pin;
 use std::time::Duration;
 use std::vec::IntoIter;
 
@@ -51,7 +52,7 @@ pub struct EventLoop {
     /// Network connection to the broker
     pub(crate) network: Option<Network>,
     /// Keep alive time
-    pub(crate) keepalive_timeout: Option<Sleep>,
+    pub(crate) keepalive_timeout: Option<Pin<Box<Sleep>>>,
     /// Handle to read cancellation requests
     pub(crate) cancel_rx: Receiver<()>,
     /// Handle to send cancellation requests (and drops)
@@ -121,7 +122,7 @@ impl EventLoop {
             self.network = Some(network);
 
             if self.keepalive_timeout.is_none() {
-                self.keepalive_timeout = Some(time::sleep(self.options.keep_alive));
+                self.keepalive_timeout = Some(Box::pin(time::sleep(self.options.keep_alive)));
             }
 
             return Ok(Event::Incoming(connack));
@@ -209,7 +210,7 @@ impl EventLoop {
             // simple. We can change this behavior in future if necessary (to prevent extra pings)
             _ = self.keepalive_timeout.as_mut().unwrap() => {
                 let timeout = self.keepalive_timeout.as_mut().unwrap();
-                timeout.reset(Instant::now() + self.options.keep_alive);
+                timeout.as_mut().reset(Instant::now() + self.options.keep_alive);
 
                 self.state.handle_outgoing_packet(Request::PingReq)?;
                 network.flush(&mut self.state.write).await?;

--- a/rumqttd/Cargo.toml
+++ b/rumqttd/Cargo.toml
@@ -20,8 +20,8 @@ path = "src/bin.rs"
 [dependencies]
 rumqttlog = { path = "../rumqttlog", version = "0.3"}
 mqtt4bytes = { path = "../mqtt4bytes", version = "0.4" }
-tokio = { version = "0.3", features = ["full"] }
-tokio-rustls = "0.21"
+tokio = { version = "1.0", features = ["full"] }
+tokio-rustls = "0.22"
 tokio-compat-02 = "0.1"
 serde = { version = "1", features = ["derive"] }
 log = "0.4"
@@ -29,7 +29,7 @@ thiserror = "1"
 argh = "0.1.3"
 confy = "0.4.0"
 pretty_env_logger = "0.4"
-bytes = "0.6"
+bytes = "1.0"
 prost = "0.6"
 warp = "0.2"
 pprof = { version = "0.3", features = ["flamegraph", "protobuf"] }

--- a/rumqttlog/Cargo.toml
+++ b/rumqttlog/Cargo.toml
@@ -10,7 +10,7 @@ description = "kafka inspired rumqtt's mqtt commitlog"
 [dependencies]
 byteorder = "1"
 memmap = "0.7"
-bytes = "0.6"
+bytes = "1.0"
 mqtt4bytes = { path = "../mqtt4bytes", version = "0.4" }
 serde = { version = "1", features = ["derive", "rc"] }
 segments = "0.1"

--- a/rumqttlog/src/router/router.rs
+++ b/rumqttlog/src/router/router.rs
@@ -845,11 +845,6 @@ mod test {
         assert!(router.readyqueue.is_empty());
     }
 
-    fn add_new_replica_connection(router: &mut Router, id: usize) {
-        let (connection, _rx) = Connection::new_replica(id, true, 10);
-        router.handle_new_connection(connection);
-    }
-
     fn add_new_remote_connection(router: &mut Router, client_id: &str) -> Receiver<Notification> {
         let (connection, rx) = Connection::new_remote(client_id, true, 10);
         router.handle_new_connection(connection);

--- a/rumqttlog/tests/mesh.rs
+++ b/rumqttlog/tests/mesh.rs
@@ -31,8 +31,8 @@ fn three_node_mesh() {
     thread::sleep(Duration::from_secs(10));
 }
 
-// TODO not workin with core_threads = 1. Investigate
-#[tokio::main(core_threads = 2)]
+// TODO: Not working with worker_threads = 1. Investigate.
+#[tokio::main(worker_threads = 2)]
 async fn node_1() {
     let config = config(1);
     let (router, router_tx) = Router::new(config);
@@ -55,7 +55,7 @@ async fn node_1() {
     }
 }
 
-#[tokio::main(core_threads = 2)]
+#[tokio::main(worker_threads = 2)]
 async fn node_2() {
     let config = config(2);
     let (router, router_tx) = Router::new(config);
@@ -87,7 +87,7 @@ async fn receive_ack(this_rx: &Receiver<RouterOutMessage>, pkid: u64) {
 }
 
 async fn wait_for_new_data(rx: &Receiver<RouterOutMessage>) -> DataReply {
-    tokio::time::delay_for(Duration::from_secs(1)).await;
+    tokio::time::sleep(Duration::from_secs(1)).await;
     match rx.try_recv() {
         Ok(RouterOutMessage::DataReply(reply)) => reply,
         v => panic!("Expecting Data Reply. {:?}", v),

